src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AutoCrafterListener.java

```diff
@@ -58,6 +67,21 @@ public void onInteract(PlayerRightClickEvent e) {
                     return;
                 }
 
+                // Check if the recipe of the item is disabled.
+                for (Recipe recipe : Bukkit.getRecipesFor(e.getItem())) {
+                    recipeKey = ((Keyed) recipe).getKey();
```

Comment:
I think it should be like that:
```suggestion
                    NamespacedKey recipeKey = ((Keyed) recipe).getKey();
```
and upper variable should be removed.

Reply:
Hmm... But it's in a for loop. Wouldn't the variable just be declared more than one time and stuff just go wrong is the for loop gets executed more than one time though?

Reply:
But variable is assigned to this loop. Should be inside.

Reply:
If you don't want you can also change if statement under this variable to this:
`if (!e.getPlayer().hasDiscoveredRecipe(((Keyed) recipe).getKey()))`

Reply:
Oh yeah true. I will change it.